### PR TITLE
feat(processors) Add http fields to event/transaction/errors processor

### PR DESCRIPTION
**Pull request by evanh**
Standardize how the http fields are parsed in the processors and add the
http method and referer to the columns.

This also removes the request field from the contexts object in the errors
processor, but all the data that was in the context is now either in a column or
in the tags.

### CODE REVIEW

File: tests/migrations/test_runner.py [hunk 1]
```diff
@@ -190,21 +190,21 @@ def get_sampling_key() -> str:
     )
 
     assert get_sampling_key() == ""
-    generate_transactions(5)
```
Comment (fpacifici) [inline tests/migrations/test_runner.py:193]: Why this change ? This is meant to test a data migration happening in `0002_transactions_onpremise_fix_orderby_and_partitionby`

Comment (evanh) [reply tests/migrations/test_runner.py:193]: Oh I see. OK well the test was failing because the method and referer fields were missing. I figured the correct way to fix that was to run the migrations, but I'll change it back.

Comment (evanh) [reply tests/migrations/test_runner.py]: OK I changed this in a way that I think makes sense.

File: tests/migrations/test_runner.py [hunk 2]
```diff
@@ -197,13 +199,13 @@ def get_sampling_key() -> str:
...
-        ),
-        force=True,
-    )
+    transaction_migrations = get_group_loader(
+        MigrationGroup.TRANSACTIONS
+    ).get_migrations()
+    for migration in transaction_migrations[1:]:
+        runner.run_migration(
+            MigrationKey(MigrationGroup.TRANSACTIONS, migration), force=True,
+        )
```
Comment (lynnagara) [inline tests/migrations/test_runner.py:208]: Why make this change? Since we're only testing the 2nd migration which is the complex one, it shouldn't depend on running the subsequent ones as well

Comment (evanh) [reply tests/migrations/test_runner.py:208]: Sorry this was a hold out when I changed the test in an earlier iteration, I'll revert.

**Review by fpacifici — APPROVED** [review]
